converter/converters/koki_service_to_kube_v1_service.go

```diff
@@ -0,0 +1,196 @@
+package converters
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/koki/short/types"
+	"github.com/koki/short/util"
+)
+
+func Convert_Koki_Service_To_Kube_v1_Service(service *types.ServiceWrapper) (*v1.Service, error) {
+	var err error
+	kubeService := &v1.Service{}
+	kokiService := &service.Service
+
+	kubeService.Name = kokiService.Name
+	kubeService.Namespace = kokiService.Namespace
+	kubeService.APIVersion = kokiService.Version
+	kubeService.Kind = "Service"
+	kubeService.ClusterName = kokiService.Cluster
+	kubeService.Labels = kokiService.Labels
+	kubeService.Annotations = kokiService.Annotations
+
+	if len(kokiService.ExternalName) > 0 {
+		kubeService.Spec.Type = v1.ServiceTypeExternalName
+		kubeService.Spec.ExternalName = kokiService.ExternalName
+		return kubeService, nil
+	}
+
+	kubeService.Spec.Type = v1.ServiceTypeClusterIP
+
+	kubeService.Spec.Selector = kokiService.PodLabels
+	kubeService.Spec.ClusterIP = string(kokiService.ClusterIP)
+	kubeService.Spec.ExternalIPs = revertExternalIPs(kokiService.ExternalIPs)
+	err = revertSessionAffinityInto(kokiService.ClientIPAffinity, &kubeService.Spec)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeService.Spec.PublishNotReadyAddresses = kokiService.PublishNotReadyAddresses
+
+	kubeService.Spec.ExternalTrafficPolicy, err = revertExternalTrafficPolicy(kokiService.ExternalTrafficPolicy)
+	if err != nil {
+		return nil, err
+	}
+
+	err = revertPortsInto(kokiService, &kubeService.Spec)
+	if err != nil {
+		return nil, err
+	}
+
+	if kokiService.LoadBalancer != nil {
+		kubeService.Spec.Type = v1.ServiceTypeLoadBalancer
+		revertLoadBalancerInto(kokiService.LoadBalancer, kubeService)
+	}
+
+	return kubeService, nil
+}
+
+func revertPort(name string, kokiPort *types.ServicePort) (*v1.ServicePort, error) {
+	kubePort := &v1.ServicePort{}
+	kubePort.Port = kokiPort.Expose
+	kubePort.TargetPort = kokiPort.PodPort
+
+	if len(name) > 0 {
+		kubePort.Name = name
+	}
+
+	if kokiPort.NodePort != 0 {
+		kubePort.NodePort = kokiPort.NodePort
+	}
+
+	if kokiPort.Protocol != "" {
+		switch kokiPort.Protocol {
+		case types.ProtocolTCP:
+			kubePort.Protocol = v1.ProtocolTCP
+		case types.ProtocolUDP:
+			kubePort.Protocol = v1.ProtocolUDP
+		default:
+			return nil, fmt.Errorf(
```

Comment:
why aren't you using util.TypeValueErrorf() everywhere?

We should use consistent error messaging.

Reply:
I'll go through the errors and change the ones that fit to util.TypeValueErrorf. I remember that a bunch of them didn't feel like a good fit, but I'll add an appropriate format to util for those.